8Knot/pages/repo_overview/visualizations/package_version.py

```diff
@@ -124,21 +124,31 @@ def process_data(df: pd.DataFrame):
     # get max time difference to use for grouping and round for correct format
     max_time = round(df["libyear"].max(), 3)
 
-    # group by the 4 buckets
-    depend_pie = pd.DataFrame(
-        df["libyear"].groupby(pd.cut(df["libyear"], [0, 0.001, 0.5, 1, max_time], right=False)).count()
-    ).rename(columns={"libyear": "packages"})
+    # create intervals for df grouper
+    intervals = [0.0, 0.001, 0.5, 1.0, max_time]
+    intervals = [val for val in intervals if val <= max_time]
+
+    # round up for formating and corner case ease
+    intervals[-1] = intervals[-1] + 0.001
+
+    # group by the time intervals
+    depend_pie = pd.DataFrame(df["libyear"].groupby(pd.cut(df["libyear"], intervals, right=False)).count()).rename(
```

Comment:
maybe add a note comment for clarity-

"groups the 'ibyear' column values by non-overlapping intervals of time, specified in the intervals object."